.github/workflows/on-pr.yml

```diff
@@ -16,3 +16,49 @@ jobs:
     uses: ./.github/workflows/build-device.yml
     with:
       timeout: 10
+
+  pre-commit:
+    name: Run Pre-commit Hooks and Propose Fixes
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history so 'origin/main' is available
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Run Pre-commit and Fix Issues
+        uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: "--from-ref origin/main --to-ref HEAD"
+        continue-on-error: false
+
+      # Declarations from here irrelevant
+      # We are failing PRs on pre-commit error
+      # Nothing below this comment will have impact
+      # Leaving this incase we want this feature later
```

Comment:
I don't get it. We will run all these steps in case the pre-commit passes? Why?

Reply:
No, but I just deleted the noise to make it more clear what happens.